File: src/learning_service/src/turtle_spawn.cpp
#include<ros/ros.h>
#include<turtlesim/Spawn.h>
int main(int argc,char** argv)
{
//初始化ROS节点
ros::init(argc,argv,"turtle_spawn");
//创建节点句柄
ros::NodeHandle node;
//创建一个服务客户端
ros::service::waitForService("/spawn");
ros::ServiceClient add_turtle=node.serviceClient<turtlesim::Spawn>("/spawn");

turtlesim::Spawn srv;
srv.request.x=2.0;
srv.request.y=2.0;
srv.request.name="turtle2";
ROS_INFO("call service to spwan turtle[x:%0.6f,y:%0.6f,name:#s]",srv.request.x,srv.request.y,srv.request.name.c_str());
add_turtle.call(srv);
ROS_INFO("Spwan turtle successfully [name:%s]",srv.response.name.c_str());
return 0;
};

File: src/learning_service/scripts/turtle_spawn.py
import sys
import rospy
from turtlesim.srv import Spawn
def turtle_spawn():
	rospy.init_node('turtlr_spawn')
	rospy.wait_for_service('/spawn')
	try:
		add_turtle = rospy.ServiceProxy('/spawn',Spawn)
		response=add_turtle(2.0,2.0,0.0,"turtle2")
		return response.name
	except rospy.ServiceException,e;
		print("service call failed:%s"%e)
if __name__=="__main__":
	print("spawn turtle successfully [name:%s]"%(turtle_spawn())
